tests/foreman/destructive/test_capsulecontent.py

```diff
@@ -0,0 +1,94 @@
+"""Capsule-Content related tests, which require destructive Satellite
+
+:Requirement: Capsule-Content
+
+:CaseAutomation: Automated
+
+:CaseLevel: System
+
+:CaseComponent: Capsule-Content
+
+:team: Phoenix-content
+
+:TestType: Functional
+
+:CaseImportance: High
+
+:Upstream: No
+"""
+import pytest
+from fauxfactory import gen_alpha
+
+from robottelo import constants
+from robottelo.api.utils import enable_rhrepo_and_fetchid
+
+pytestmark = [pytest.mark.destructive]
+
+
+@pytest.mark.tier4
+@pytest.mark.skip_if_not_set('capsule')
+def test_positive_sync_without_deadlock(
+    target_sat, large_capsule_configured, function_entitlement_manifest_org
+):
+    """Synchronize one bigger repo published in multiple CVs to a blank Capsule.
+    Assert that the sync task succeeds and no deadlock happens.
+
+    :id: 91c6eec9-a582-46ea-9898-bdcaebcea2f0
+
+    :setup:
+        1. A blank external capsule that has not been synced yet (!) with immediate download
+           policy and running multiple (4 and more) pulpcore workers.
+
+    :steps:
+        1. Sync one bigger repository to the Satellite.
+        2. Create a Content View, add the repository and publish it.
+        3. Create several copies of the CV and publish them.
+        4. Add the Library LCE to the Capsule.
+        5. Sync the Capsule.
+
+    :expectedresults:
+        1. Sync passes without deadlock.
+
+    :customerscenario: true
+
+    :BZ: 2062526
+
+    """
+    # Note: As of now BZ#2122872 prevents us to use the originally intended RHEL7 repo because
+    # of a memory leak causing Satellite OOM crash in this scenario. Therefore, for now we use
+    # smaller RHSCL repo instead, which was also capable to hit the deadlock issue, regardless
+    # the lower rpms count. When the BZ is fixed, reconsider upscale to RHEL7 repo or similar.
+    repo_id = enable_rhrepo_and_fetchid(
```

Comment:
@vsedmik Yes, you're right, enable_rhrepo_and_fetchid might be running on default sat instead of destructive sat
```suggestion
    repo_id = target_sat.api_factory.enable_rhrepo_and_fetchid(
```

Reply:
Well, we don't have that in `api_factory` in 6.12.
Although it's probably out of the scope of this cherry-pick, let me check if I can pull it into 6.12.

Reply:
Looks like the `api_factory` update works and PRT passed @Gauravtalreja1